src/lib.rs

```diff
@@ -307,6 +307,14 @@ impl Body for String {
     ) -> Poll<Result<Option<HeaderMap>, Self::Error>> {
         Poll::Ready(Ok(None))
     }
+
+    fn is_end_stream(&self) -> bool {
+        self.is_empty()
```

Comment:
I know very little about the underlying protocols but is an empty string the same as EOS? I dunno if sending one empty frame is the same being done sending stuff, or if that question even makes sense.

Reply:
Ah I see you return `None` in `poll_data` if the buf is empty. So thats probably fine :)

Reply:
Well when the string is empty it matches this, right?

     /// An end of stream means that both `poll_data` and `poll_trailers` will
    /// return `None`.

Reply:
Yep all good!